pubtools/_quay/push_docker.py

```diff
@@ -365,68 +368,51 @@ def generate_backup_mapping(
+                                arch_mads = [
+                                    m
+                                    for m in cast(List[ManifestArchDigest], man_arch_digs)
+                                    if m.arch == "amd64"
+                                ]
+                            else:
+                                arch_mads = cast(List[ManifestArchDigest], man_arch_digs)
+                            v2list_mads: List[ManifestArchDigest | None] = cast(
+                                List[ManifestArchDigest | None],
+                                [
+                                    m
+                                    for m in cast(List[ManifestArchDigest], man_arch_digs)
+                                    if m.type_ == QuayClient.MANIFEST_LIST_TYPE
+                                ]
+                                or [None],
+                            )
+                            v2s1_mads: List[ManifestArchDigest | None] = cast(
+                                List[ManifestArchDigest | None],
+                                [m for m in arch_mads if m.type_ == QuayClient.MANIFEST_V2S1_TYPE]
+                                or [None],
+                            )
+                            v2s2_mads: List[ManifestArchDigest | None] = cast(
+                                List[ManifestArchDigest | None],
+                                [m for m in arch_mads if m.type_ == QuayClient.MANIFEST_V2S2_TYPE]
+                                or [None],
+                            )
+                            for mads, digest_mask in zip(
+                                (v2s1_mads, v2s2_mads, v2list_mads), (1, 2, 3)
                             ):
-                                if mad:
-                                    image_data = PushDocker.ImageData(
-                                        full_repo,
-                                        d_tag,
-                                        (
-                                            cast(ManifestArchDigest, v2list_mad).digest
-                                            if digest_mask == 3
-                                            else ""
-                                        ),
-                                        (
-                                            cast(ManifestArchDigest, v2s2_mad).digest
-                                            if digest_mask == 2
-                                            else ""
-                                        ),
-                                        (
-                                            cast(ManifestArchDigest, v2s1_mad).digest
-                                            if digest_mask == 1
-                                            else ""
-                                        ),
-                                    )
-                                    backup_tags[image_data] = json.loads(mad.manifest)
+                                if mads:
```

Comment:
Is this condition necessary if mads is empty? Or is it here because empty list is actually [None]? Why is that the case?

Reply:
yeah, you're right, it's probably redundant